src/context/UserProvider.tsx

```diff
@@ -51,15 +51,24 @@ const UserContext = createContext<IUserContext>({
 const apolloClient = initializeApollo();
 
 export const UserProvider = (props: { children: ReactNode }) => {
-	const { account, library, chainId } = useWeb3React();
-	useWallet();
+	const { account, library, chainId, activate } = useWeb3React();
 
 	const [user, setUser] = useState<IUser | undefined>();
 	const [balance, setBalance] = useState<string | null>(null);
 
 	const isEnabled = !!library?.getSigner() && !!account && !!chainId;
 	const isSignedIn = isEnabled && !!user?.token;
 
+	useEffect(() => {
+		const selectedWalletName = localStorage.getItem(StorageLabel.WALLET);
+		const wallet = walletsArray.find(w => w.value === selectedWalletName);
+		if (wallet) {
+			activate(wallet.connector, err => {
+				console.log('err', err);
+			});
+		}
+	}, [activate]);
```

Comment:
`activate` is a always a function and I think adding it as a dependency is not necessary. If user locks his wallet and unlock again, the app won't try to `activate`, So I think you should replace `activate` dependency with web3React `active`

Reply:
I put this for the time that the activate function changed, for example, if it is not initial yet.